chctools/btor2.py

```diff
@@ -849,11 +833,72 @@ def __init__(self, nid: int, n: int, expr_list: List[Expr], symbol: str = ""):
         self.expr_list = expr_list
 
 
+class Ts:
+    pre_vars: List[z3.ExprRef]
+    post_vars: List[z3.ExprRef]
+    inputs: List[z3.ExprRef]
+    init: z3.ExprRef
+    tr: z3.ExprRef
+    bad: z3.ExprRef
+
+    def __init__(self):
+        self.pre_vars = []
+        self.post_vars = []
+        self.inputs = []
+        self.init = z3.BoolVal(True)
+        self.tr = z3.BoolVal(True)
+        self.bad = z3.BoolVal(False)
+
+    def put_var(self, sort: z3.SortRef, prefix: str = "") -> Tuple[z3.ExprRef, z3.ExprRef]:
+        pre_z3_expr: z3.ExprRef = z3.FreshConst(sort, "v" + prefix)
+        post_z3_expr: z3.ExprRef = z3.FreshConst(sort, "v_out" + prefix)
+        self.pre_vars.append(pre_z3_expr)
+        self.post_vars.append(post_z3_expr)
+        return pre_z3_expr, post_z3_expr
+
+    def put_input(self, sort: z3.SortRef, prefix: str = "") -> z3.ExprRef:
+        input_z3_expr: z3.ExprRef = z3.FreshConst(sort, "i" + prefix)
+        self.inputs.append(input_z3_expr)
+        return input_z3_expr
+
+    def sig(self) -> List[z3.SortRef]:
+        return [v.sort() for v in self.post_vars]
+
+    def inv(self) -> z3.FuncDeclRef:
+        return z3.Function("Inv", *(self.sig() + [z3.BoolSort()]))
+
+    def vars(self) -> List[z3.ExprRef]:
+        return self.pre_vars + self.post_vars
+
+    def all(self) -> List[z3.ExprRef]:
+        return self.vars() + self.inputs
+
+    def to_chc(self) -> List[str]:
```

Comment:
would be useful to factor this method out so that transition system does not determine how chc are produced

would also be useful to have an implementation that uses current `HornDb` interface rather than generate strings directly. The string-based method is still useful for testing, so it is ok to keep it here as well.